sycl/include/CL/sycl/detail/program_impl.hpp

```diff
@@ -50,14 +50,9 @@ class program_impl {
   // caching resolution
   program_impl(vector_class<std::shared_ptr<program_impl>> ProgramList,
                string_class LinkOptions = "")
-      : State(program_state::linked), LinkOptions(LinkOptions),
+      : State(program_state::linked), Context(ProgramList.at(0)->Context),
+        Devices(ProgramList.at(0)->Devices), LinkOptions(LinkOptions),
         BuildOptions(LinkOptions) {
-    // Verify arguments
-    if (ProgramList.empty()) {
-      throw runtime_error("Non-empty vector of programs expected");
```

Comment:
Why this check is removed?

Reply:
vector::at(0) is used instead. It does the check and generates std::out_of_range if vector is empty.
From my perspective it is more or less equal.

Reply:
I think old exception described the problem better.

Reply:
but it does not allow to avoid unused object creation.

Reply:
@romanovvlad , please provide your opinion on this.

Reply:
I tend to agree with @Fznamznon . It's better to use sycl type of exception with more clear message here.